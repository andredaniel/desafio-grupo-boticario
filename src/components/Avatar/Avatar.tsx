import React, { ImgHTMLAttributes } from 'react'
import { AvatarWrapper } from './Avatar.styles'

const Avatar: React.FC<ImgHTMLAttributes<HTMLImageElement>> = (
  props
): JSX.Element => {
  const { src } = props

  return <AvatarWrapper src={src} />
}

export { Avatar }
