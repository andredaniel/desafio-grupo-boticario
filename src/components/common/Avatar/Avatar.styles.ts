import styled from 'styled-components'

export const AvatarWrapper = styled.img`
  border-radius: 50%;
  height: 40px;
  width: 40px;
`
