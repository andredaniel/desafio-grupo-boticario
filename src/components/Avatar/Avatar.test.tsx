import { render } from '@testing-library/react'
import React from 'react'
import { Avatar } from './Avatar'

describe('<Avatar />', () => {
  it('should render an Avatar with image', () => {
    const { container } = render(<Avatar src="//placehold.it/200/200" />)
    expect(container).toMatchSnapshot()
  })

  it('should render an Avatar without image', () => {
    const { container } = render(<Avatar />)
    expect(container).toMatchSnapshot()
  })
})
