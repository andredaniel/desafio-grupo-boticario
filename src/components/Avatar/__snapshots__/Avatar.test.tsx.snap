// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Avatar /> should render an Avatar with image 1`] = `
<div>
  <img
    class="sc-bdfBwQ ehnBFF"
    src="//placehold.it/200/200"
  />
</div>
`;

exports[`<Avatar /> should render an Avatar without image 1`] = `
<div>
  <img
    class="sc-bdfBwQ ehnBFF"
  />
</div>
`;
